metadata/dashboard/dashboard_metadata.go

```diff
@@ -923,22 +927,44 @@ func (m *MetadataServer) GetMetadataList(c *gin.Context) {
 
 }
 
+func (m *MetadataServer) GetSearch(c *gin.Context) {
+	query := c.Param("query")
+	searchParameters := &api.SearchCollectionParams{
+		Q:       query,
+		QueryBy: "Name",
+	}
+	result, err := typesenseClient.Collection("resource").Documents().Search(searchParameters)
+	if err != nil {
+		c.JSON(500, "Failed to fetch resources")
+	}
+	c.JSON(200, result)
+}
+
 func (m *MetadataServer) Start(port string) {
 	router := gin.Default()
 	router.Use(cors.Default())
 
 	router.GET("/data/:type", m.GetMetadataList)
 	router.GET("/data/:type/:resource", m.GetMetadata)
+	router.GET("/search/:query", m.GetSearch)
 
 	router.Run(port)
 }
 
 func main() {
+	logger := zap.NewExample().Sugar()
 	metadataHost := os.Getenv("METADATA_HOST")
 	metadataPort := os.Getenv("METADATA_PORT")
+	typesenseHost := os.Getenv("TYPESENSE_HOST")
+	typesensePort := os.Getenv("TYPESENSE_PORT")
+	typesenseEndpoint := fmt.Sprintf("http://%s:%s", typesenseHost, typesensePort)
+	typesenseApiKey := os.Getenv("TYPESENSE_APIKEY")
+	logger.Infof("Connecting to typesense at: %s", typesenseEndpoint)
+	typesenseClient = typesense.NewClient(
+		typesense.WithServer(typesenseEndpoint),
+		typesense.WithAPIKey(typesenseApiKey))
 	metadataAddress := fmt.Sprintf("%s:%s", metadataHost, metadataPort)
-	fmt.Println("Looking for metadata at: ", metadataAddress)
-	logger := zap.NewExample().Sugar()
+	logger.Infof("Looking for metadata at: %s", metadataAddress)
 	client, err := metadata.NewClient(metadataAddress, logger)
 	if err != nil {
 		logger.Panicw("Failed to connect", "Err", err)
```

Comment:
Not sure if this was me or you but can you change Err -> error

Reply:
changed